components/molecule/modal/src/HoC/WithUrlState.js

```diff
@@ -0,0 +1,76 @@
+import React, {useEffect, useState} from 'react'
+import PropTypes from 'prop-types'
+
+const getUrlWithHash = hashName => {
+  const location = window.location
+  const hash = hashName ? `#${hashName}` : ''
+  return `${location.origin}${location.pathname}${location.search}${hash}`
+}
+
+const checkHash = hash => window.location.hash.includes(hash)
+
+const emptyFn = () => {}
+
+/**
+ * It must be refactored with react-router hooks (in React Router > 5)
+ * - useHistory
+ * - useLocation
+ */
+export default BaseComponent => {
+  const displayName = BaseComponent.displayName
+
+  if (typeof window === 'undefined') return BaseComponent
+
+  const WithUrlState = props => {
+    const {hash, openModalTrigger, onClose, ...rest} = props
+    const {isOpen} = rest
+    const [isPopStateChange, setIsPopStateChange] = useState(checkHash(hash))
+
+    const _handlePopState = ev => {
+      const hasHash = checkHash(hash)
+      setIsPopStateChange(true)
+
+      if (hasHash) openModalTrigger()
+      else onClose()
+    }
+
+    const _handleChangeState = () => {
+      if (isPopStateChange) return setIsPopStateChange(false)
+
+      const nextUrl = isOpen ? getUrlWithHash(hash) : getUrlWithHash()
+
+      window.history.pushState(
```

Comment:
This is less than 80 chars, make it be in 1 line manually and once you save, this will stay in one line I think